packages/logger/README.md

```diff
@@ -1,4 +1,320 @@
+  },
+  paths:{
+      ...
+  }
+}
+```
+Also note that LoopBack Next provides `validateApiSpec` in the testlab so that you can validate it in `npm test` of your app.  Isn't that cool?
+
+
+
+# LoopBack Next
+
+We're going to examine three key concepts: controller & apiSpec, components & providers, and custom sequence.  Working sample codes are available in several places, one of which is [Hello World](https://github.com/strongloop/loopback-next-hello-world).  I'm going to focus on specific conceptual roadblocks I stumbled on.
+
+Skeleton of the client code looks like this:
+```
+  const app = new Application({
+    components: [HisComponent],
+   });
+  app.controller(HelloWorldController);
+  app.sequence(MySequence);
+```
+
+### Controllers and API Spec
+
+Let's take a look at HelloWorldController code and associated API spec first, then we study what they do.
+
+```
+import {api inject} from '@loopback/core';
+import {apiSpec} from './hello-world.api';
+
+@api(apiSpec)
+export class HelloWorldController {
+  @inject('defaultName');
+  private name: string;
+  constructor() {}
+
+  helloWorld(name?: string) {
+      name = name || this.name;
+      return `Hello world ${name}!`
+  }
+}
+
+```
+```
+// hello-world.api.ts
+export const apiSpec =
+{
+  "paths": {
+    "/helloworld": {
+      "get": {
+        "x-operation-name": "helloWorld",
+        "parameters": [
+          {
+            "name": "name",
+            "in": "query",
+            "description": "Your name.",
+            "required": false,
+            "type": "string"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Returns a hello world with your (optional) name."
+          }
+        }
+      }
+    }
+  }
+}
+```
+@api is a [class decorator](https://www.typescriptlang.org/docs/handbook/decorators.html).  The class decorator,
+`api` is a function that is called when the class, HelloWorldController is instantiated.  In this case, `api` registers
+the API specification to the application.  The decorator site describes some relatively complex concept, but we can just read the class decorator section to get a high-level understanding of decorator.  The concept is the same for other types of decorators.  That's enough.
+
+In the above sample code, @inject is a property decorator defined by LoopBack Next.  It's a very important LoopBack Next concept.  It works this way: @inject acquires the value bound to the key: `defaultName` in the application's context, then assigns the value to `name` property when the class `HelloWorldControlle` is instantiated.
+
+Please note that we introduced another new term, `Context`.  Context is simply a memory space where LoopBack Next maintains key-value pairs.  There are two types of context: application context and request context.  In this blog, we deal with only application context.
+
+The LoopBack Next specific decorator, @inject can be used as a parameter decorator e.g.:
```

Comment:
Yes and no. While decorators can be used for parameters in general, the `@inject` decorator is supported only for class constructor parameters.

```ts
class MyClass {
  constructor(
    @inject('defaultName') name: string
  } {}
}
```